exercises/diffie-hellman/description.md

```diff
@@ -28,11 +28,11 @@ private key b.
 
 Alice and Bob exchange public keys.  Alice calculates secret key s.
 
-    s = B**a mod p
+    s = B^a mod p
 
 Bob calculates
 
-    s = A**b mod p
+    s = A^b mod p
 
 The calculations produce the same result!  Alice and Bob now share
 secret s.
```

Comment:
I would like to suggest adding a note be added at the end of this document that mentions that in some programming languages `**` is used as the method to indicate exponentiations instead of `^`.

I think just replacing one form for another doesn't necessarily clear up the confusion for everyone, just shifts it.

Reply:
Might it be better to remove any implementation dependent form and instead use the mathematical form? That is what is being described after all.

Reply:
Hard agree with @wolf99 comment.  For example, in Python, `^` is used as a `set` operator, which is why `**` is used for exponentiation.  It varies widely by programming language.

Reply:
@wolf99 I like your suggestion, however I worry about how the special character will translate for users.  If there is no concern to worry about then 👍 from me too.

Reply:
I feel exactly like @rpottsoh : would `aᵇ` render properly in any IDE for people downloading the exercise through CLI? 


> I would like to suggest adding a note be added at the end of this document that mentions that in some programming languages `**` is used as the method to indicate exponentiations instead of `^`.

Well, the note might just mention that a^b means "a raised to the power of b", or "a*a*a*...*a" b times", so that the description only uses mathematics and no programming conventions?

Reply:
We have other exercises that use them. If an IDE can't display standard characters like that, it's most likely a deliberate choice by the student to use archaic IDE and make their life hard.

Reply:
Thus, I see no reason not to use this notation. I'll make a new commit.

Reply:
> deliberate choice by the student to use archaic IDE

I use `ed` (not always) and it displays it, so I wonder if you mean "IDE that sucks" rather than archaic.

Reply:
Instead of trying to abstract it away:

```text
s = (A to the power of b) mod p
 ```

Reply:
That is far more abstract and less readable